app/src/main/java/com/firebase/uidemo/auth/AuthUiActivity.java

```diff
@@ -97,15 +105,18 @@
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        FirebaseAuth auth = FirebaseAuth.getInstance();
-        if (auth.getCurrentUser() != null) {
-            startActivity(SignedInActivity.createIntent(this));
-            finish();
-        }
-
         setContentView(R.layout.auth_ui_layout);
         ButterKnife.bind(this);
 
+        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
+        if (user != null) {
+            if (!user.isAnonymous()) {
+                startActivity(SignedInActivity.createIntent(this));
```

Comment:
Let's discuss changes to the sample last, I want to focus on the changes to the library at first since that will have effects on how the sample is structured.